src/amr/data/field/coarsening/electric_field_coarsener.hpp

```diff
@@ -55,78 +53,92 @@ class MagneticFieldCoarsener
 
         core::Point<int, dimension> fineStartIndex;
 
-        fineStartIndex[dirX] = coarseIndex[dirX] * this->ratio_;
-
-        if constexpr (dimension > 1)
+        for (auto i = std::size_t{0}; i < dimension; ++i)
         {
-            fineStartIndex[dirY] = coarseIndex[dirY] * this->ratio_;
-            if constexpr (dimension > 2)
-            {
-                fineStartIndex[dirZ] = coarseIndex[dirZ] * this->ratio_;
-            }
+            fineStartIndex[i] = coarseIndex[i] * this->ratio_;
```

Comment:
1. `ratio_` is static, so there is no `this`
2. you should probably be using the constant in `amr/amr_constants.hpp`